### ISSUE TITLE
Used Without Corresponding Plugin

### ISSUE THREAD

**Issue by lloy0076** [shakacode/bootstrap-loader#232]
I keep getting:

```
ERROR in ./~/extract-text-webpack-plugin/loader.js?{"omit":1,"remove":true}!./~/style/style.js!./~/css-loader!./~/resolve-url-loader!./~/sass-loader?sourceMap!./~/bootstrap-loader/lib/bootstrap.styles.loader.js!./~/bootstrap-loader/no
js                                                                                                                                                                                                                                        
Module build failed: Error: "extract-text-webpack-plugin" loader is used without the corresponding plugin, refer to https://github.com/webpack/extract-text-webpack-plugin for the usage example                                          
    at Object.module.exports.pitch (c:\laragon\www\foo-bar\node_modules\extract-text-webpack-plugin\loader.js:26:9)                                                                                                                       
 @ ./~/bootstrap-loader/lib/bootstrap.loader.js?extractStyles!./~/bootstrap-loader/no-op.js 1:21-237                                                                                                                                      
 @ ./~/bootstrap-loader/extractStyles.js                                                                                                                                          ```
I evilly forced the check to NOT fail (by doing "if (false && ...)" and got:

> ERROR in ./~/extract-text-webpack-plugin/loader.js?{"omit":1,"remove":true}!./~/style/style.js!./~/css-loader!./~/resolve-url-loader!./~/sass-loader?sourceMap!./~/bootstrap-loader/lib/bootstrap.styles.loader.js!./~/bootstrap-loader/no-op.
> js                                                                
> Module build failed: TypeError: this[NS] is not a function  at Object.module.exports.pitch (c:\laragon\www\foo-bar\node_modules\extract-text-webpack-plugin\loader.js:32:20)                                                                                                                          
>  @ ./~/bootstrap-loader/lib/bootstrap.loader.js?extractStyles!./~/bootstrap-loader/no-op.js 1:21-237                                                                                                                                          
>  @ ./~/bootstrap-loader/extractStyles.js            
                                                                                                                          ```

With:

```
{                                                 
{                                                 
  "name": "foo-bar",                              
  "version": "1.0.0",                             
  "description": "A foo bar project.",            
  "main": "index.js",                             
  "author": "David Lloyd <lloy0076@adam.com.au>", 
  "license": "Apache-2.0",                        
  "dependencies": {                               
    "autoprefixer": "^6.6.1",                     
    "babel-core": "^6.21.0",                      
    "babel-loader": "^6.2.10",                    
    "babel-preset-latest": "^6.16.0",             
    "bootstrap-loader": "^2.0.0-beta.18",         
    "bootstrap-sass": "^3.3.7",                   
    "css-loader": "*",                            
    "cssnano": "^3.9.1",                          
    "exports-loader": "^0.6.3",                   
    "extract-text-webpack-plugin": "2.0.0-beta.4",
    "file-loader": "^0.9.0",                      
    "imports-loader": "^0.7.0",                   
    "node-sass": "*",                             
    "postcss-loader": "^1.2.1",                   
    "resolve-url-loader": "*",                    
    "sass-loader": "*",                           
    "style": "^0.0.3",                            
    "style-loader": "^0.13.1",                    
    "url-loader": "*",                            
    "webpack": "2.1.0-beta.19"                    
  }                                               
}                                                                                          
```

On:

```
λ node -v
v6.9.2   

λ webpack --help        
webpack 2.1.0-beta.19
```

I fiddled the error message to see what might be missing and it's a 'this[NS]' missing - I made it say what NS was:

```
Module build failed: Error: "NS is "c:\laragon\www\foo-bar\node_modules\extract-text-webpack-plugin" --- "extract-text-webpack-plugin" loader is used without the corresponding plugin, refer to https://github.com/webpack/extract-text-webpa
ck-plugin for the usage example                                                                                                                                                            ```

So this["c:\laragon\www\foo-bar\node_modules\extract-text-webpack-plugin"] is apparently undefined/null.

Not sure what is happening here.

**Comment by lloy0076** [comment]
(sorry re: formatting - the markdown is really not helpful)

**Comment by lloy0076** [comment]
Aha!

Ok, this is a "my fault".

Closing.

**Comment by mmahalwy** [comment]
@lloy0076 would help if you tell us what 'my fault' is to help other devs

**Comment by lloy0076** [comment]
It was obvious once I spotted it!

"My fault" was that I hadn't included the extract-text-webpack-plugin in the plugins array.

Unfortunately I got confused because I was migrating from webpack 1.4 to 2.X and at the start didn't get the migration entirely correct webpack 1.4 would throw really odd errors because it got a webpack 2.X configuration.

Once I'd finally managed to ensure that the only webpack version being used was 2.X (in me case 2.1.0-beta19) the error threw me even though it was saying precisely what was wrong. Like Alice in Wonderland, I went down the rabbit hole instead of first checking the obvious fix.

It wasn't until staring at the extract-text-webpack's git documentation that I suddenly did see the problem staring at me under my nose.
